rasa/core/channels/hangouts.py

```diff
@@ -254,21 +264,22 @@ def _extract_input_channel(self) -> Text:
         return self.name()
 
     def _check_token(self, bot_token: Text) -> None:
-        # see https://developers.google.com/hangouts/chat/how-tos/bots-develop#verifying_bot_authenticity # noqa: E501, W505
+        # see https://developers.google.com/chat/how-tos/bots-develop#verifying_bot_authenticity # noqa: E501, W505
+        # and https://google-auth.readthedocs.io/en/latest/user-guide.html#identity-tokens # noqa: E501, W505
         try:
-            token = client.verify_id_token(
-                bot_token, self.project_id, cert_uri=CERT_URI
+            decoded_token = id_token.verify_token(
+                bot_token, self.google_request, audience=self.project_id
             )
+        except ValueError:
```

Comment:
Is this the only exception that could be raised?

Reply:
This and `TransportError` but [TransportError](https://github.com/googleapis/google-auth-library-python/blob/8f1e9cfd56dbaae0dff64499e1d0cf55abc5b97e/google/oauth2/id_token.py#L102) we can't really do much about. If there's a network problem, we can't fetch certs and hence we can't verify request is coming from a Google service we can't really return 401. 500 default seems like a good response in that case.